run.sh

```diff
@@ -2,6 +2,39 @@
 
 set -ex
 
+# Ensure we're running on a Raspberry Pi
+if ! grep -q "Raspberry Pi" /proc/device-tree/model 2>/dev/null; then
+	echo "This script is intended to run on a Raspberry Pi."
+	exit 1
+fi
+
+# Ensure symbolic link is created (only if it doesn't exist)
+if [ ! sudo test -L /root/.cargo ]; then
+	sudo ln -s /root/.local /root/.cargo
+fi
+
+# Check if modprobe is installed
+if ! command -v modprobe >/dev/null 2>&1; then
+	echo "modprobe not found. Installing it..."
+	
+	# Install modprobe for Raspberry Pi (Debian-based system)
+	sudo apt-get update && sudo apt-get install -y kmod
+fi
+
+# Ensure I2C is enabled
+if ! lsmod | grep -q i2c_dev; then
+	echo "I2C module not loaded. Loading i2c-dev..."
+	sudo modprobe i2c-dev
+fi
+
+
+# Ensure I2C is enabled in the Raspberry Pi config
+if ! grep -q "^dtparam=i2c_arm=on" /boot/config.txt; then
+	echo "Enabling I2C in /boot/config.txt..."
+	echo "dtparam=i2c_arm=on" | sudo tee -a /boot/config.txt
+	sudo reboot
+fi
```

Comment:
rebooting the pi is also pretty disruptive, so it might be better to just throw an error and document in the readme how to enable it. personally I'm fine with either